clients/drcachesim/tools/func_view.cpp

```diff
@@ -197,7 +213,7 @@ func_view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
     if (memref.marker.type != TRACE_TYPE_MARKER)
         return true;
     process_memref_for_markers(shard, memref);
-    return true;
+    return shard->error.empty(); // An error message means there was a problem.
```

Comment:
Clearer to have process_memref_for_markers() return the error string?

Reply:
Since this still has to be converted to a bool for the parallel_shard_memref return, I don't think returning the error string here helps.  But I changed it to return bool, and this led me to notice that process_memref() needs to detect errors, so I updated it, too.